File: workspace-service/src/main/java/com/teamcollab/workspaceservice/workspace/event/WorkspaceCreatedEvent.java
package com.teamcollab.workspaceservice.workspace.event;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class WorkspaceCreatedEvent {

    private UUID eventId;

    private String workspaceName;

    private String createdBy;

    private String createdByEmail;

    private LocalDateTime occurredOn;

}

File: notification-service/Events/WorkspaceCreatedEvent.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace NotificationService.Events
{

public class WorkspaceCreatedEvent
    {
        public Guid EventId { get; set; }

        public string EventName { get; set; }

        public string WorkspaceId { get; set; }

        public string WorkspaceName { get; set; }

        public string CreatedBy { get; set; }

        public string CreatedByEmail { get; set; }

        public DateTime OccurredOn { get; set; }
    }
}
